### PR TITLE
Make VOD playback work again.

### PR DESCRIPTION
**Pull request by fredrikaubert**
Propably some changes made in nrk api causing VOD playback to not work for me on kodi 19. Only live streams worked. This pull-request should make browsing and starting streams from categories work again.

### CODE REVIEW

**Comment by petterreinholdtsen** [comment]
[fredrikaubert]
> Propably some changes made in nrk api causing VOD playback to not work
> for me on kodi 19. Only live streams worked. This pull-request should
> make browsing and starting streams from categories work again.

Nice.  Perhaps you want to maintain the kodi package?  @tamland have
stated he no longer want to.

-- 
Happy hacking
Petter Reinholdtsen

File: nrktv.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 import datetime
 import re
+import xbmc
```
Comment (CaptainTyphoo) [inline nrktv.py:22]:
```suggestion

```
please remove the unnecessary import

File: nrktv.py [hunk 2]
```diff
@@ -138,20 +139,14 @@ def from_response(r):
...
-                            ',' + ','.join(bitrates) + ',',
-                            'master.m3u8')
-                except (KeyError, IndexError):
-                    # Fall back to previous behavior
-                    return part['hlsUrl']
-
-            parts = sorted(r['mediaAssetsOnDemand'], key=lambda x: x['part'])
-            media_urls = [format_path(part) for part in parts]
+
+        manifest = session.get('https://psapi.nrk.no/playback/manifest/{}'.format(r.get('id')))
```
Comment (CaptainTyphoo) [inline nrktv.py:143]:
```suggestion
        manifest = _get('/playback/manifest/{}'.format(r.get('id')))
```
please use the already existing function for GET requests

File: nrktv.py [hunk 3]
```diff
@@ -138,20 +139,14 @@ def from_response(r):
...
+        #xbmc.log("---------------->Manifest %s" % manifest.json(), xbmc.LOGERROR)
```
Comment (CaptainTyphoo) [inline nrktv.py:144]:
```suggestion

```
please do not add commented out code

File: nrktv.py [hunk 4]
```diff
@@ -138,20 +139,14 @@ def from_response(r):
...
+
+        if manifest.json()['playability'] == 'playable':
+            media_urls = list( map(lambda x: x['url'], manifest.json()['playable']['assets']) )
+            #xbmc.log("============> mediaurls %s" % media_urls, xbmc.LOGERROR)
```
Comment (CaptainTyphoo) [inline nrktv.py:148]:
```suggestion

```
please do not add commented out code

File: subs.py [hunk 5]
```diff
@@ -23,21 +23,26 @@
...
 
-    content = _ttml_to_srt(html)
+    content = vtt_to_srt(subs)
     filename = os.path.join(xbmc.translatePath("special://temp"), 'nor.srt')
     with open(filename, 'w' ,encoding='utf8') as f:
         f.write(content)
     return filename
 
 
+def vtt_to_srt(content):
```
Comment (CaptainTyphoo) [inline subs.py:42]: 1. Is there no conversion from vtt to srt required?

2. Please make the the function private:
```suggestion
def _vtt_to_srt(content):
```

File: subs.py [hunk 6]
```diff
@@ -23,21 +23,26 @@
...
+    return content
+
+
 def _ttml_to_srt(ttml):
```
Comment (CaptainTyphoo) [inline subs.py:46]: Please remove the _ttml_to_srt function as it is no longer used.

File: subs.py [hunk 7]
```diff
@@ -23,21 +23,26 @@
...
-    ttml_sub_url = mediaelement_json["subtitlesUrlPath"]
-    html = requests.get(ttml_sub_url).text
-    if not html:
+    ttml_sub_url = mediaelement_json["playable"]['subtitles'][0]['webVtt']
+    subs = requests.get(ttml_sub_url).text
+    if not subs:
         return None
 
-    content = _ttml_to_srt(html)
+    content = vtt_to_srt(subs)
```
Comment (CaptainTyphoo) [inline subs.py:35]:
```suggestion
    content = _vtt_to_srt(subs)
```
please make the the function private

**Review by CaptainTyphoo — CHANGES_REQUESTED** [review]
Thank you very much for your contribution!
Could you please incorporate my review comments?
I will merge your changes to master after my comments have been resolved.

File: subs.py (hunk 5 above)
Comment (fredrikaubert) [reply subs.py]: Don't really know if there are any conversion needed. I'm not really familiar with the different subtitle-formats. 

I've been running this code for some time watching nrk content, and I've never noticed any issues with the subs.

**Comment by fredrikaubert** [comment]
> Nice. Perhaps you want to maintain the kodi package? @tamland have stated he no longer want t

Thanks, but no thanks. :-) I don't have the pastime to commit to maintaining right now.

**Review by CaptainTyphoo — APPROVED** [review]
Looks good!

**Comment by andrea99boban** [comment]
I made this addon (after pull #71) to work under Kodi 20.0, few lines changed in 2 files. Is this of interest to evaluate being merged in?

**Comment by petterreinholdtsen** [comment]
[andrea99boban]
> I made this addon (after pull #71) to work under Kodi 20.0, few lines
> changed in 2 files. Is this of interest to evaluate being merged in?

I would love to see a new and working version in the Kodi add-on
repositories, at least.  But I am just a random user passing by.  I
suspect a fork is needed as the original developer @tamland have stated
he is busy elsewhere these days.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by andrea99boban** [comment]
fork done, also as pull #73 here

**Comment by solbu** [comment]
> fork done

As a service to others, could you link to your fork, making it easier to find? :-)

**Comment by solbu** [comment]
And if you plan on being the official maintainer, you should open the issues feature on your fork.

**Pull request merged**